x-pack/plugins/reporting/server/export_types/csv/generate_csv/index.ts

```diff
@@ -73,64 +63,35 @@ export function createGenerateCsv(logger: LevelLogger) {
       };
     }
 
-    const iterator = hitIterator(
-      settings.scroll,
-      callEndpoint,
-      job.searchRequest,
-      cancellationToken
-    );
     let maxSizeReached = false;
     let csvContainsFormulas = false;
 
-    const flattenHit = createFlattenHit(fields, metaFields, conflictedTypesFields);
-    const formatsMap = await getFieldFormats()
-      .fieldFormatServiceFactory(uiSettingsClient)
-      .then((fieldFormats) =>
-        fieldFormatMapFactory(job.indexPatternSavedObject, fieldFormats, settings.timezone)
-      );
-
-    const formatCsvValues = createFormatCsvValues(
-      escapeValue,
-      settings.separator,
-      fields,
-      formatsMap
-    );
-    try {
-      while (true) {
-        const { done, value: hit } = await iterator.next();
-
-        if (!hit) {
-          break;
-        }
-
-        if (done) {
-          break;
-        }
-
-        if (cancellationToken.isCancelled()) {
-          break;
-        }
+    for (const hit of table.rows) {
+      if (cancellationToken.isCancelled()) {
+        break;
+      }
 
-        const flattened = flattenHit(hit);
-        const rows = formatCsvValues(flattened);
-        const rowsHaveFormulas =
-          settings.checkForFormulas && checkIfRowsHaveFormulas(flattened, fields);
+      const rowsHaveFormulas = settings.checkForFormulas && checkIfRowsHaveFormulas(hit, fields);
 
-        if (rowsHaveFormulas) {
-          csvContainsFormulas = true;
-        }
+      if (rowsHaveFormulas) {
+        csvContainsFormulas = true;
+      }
 
-        if (!builder.tryAppend(rows + '\n')) {
-          logger.warn('max Size Reached');
-          maxSizeReached = true;
-          if (cancellationToken) {
-            cancellationToken.cancel();
-          }
-          break;
+      const columns = table.columns.map((c) => {
+        // we are working on this utilities, its not there yet but should be in a week or so, which will get us the formatted values
```

Comment:
@ppisljar will you be able to update this PR in the near future? This TODO should be taken care of by now

Reply:
yes i will continue working on this (after new year)